src/examples/src/widgets/menu/MemoryResource.tsx

```diff
@@ -0,0 +1,44 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import List, { defaultTransform } from '@dojo/widgets/list';
+import icache from '@dojo/framework/core/middleware/icache';
+import { createResource, DataTemplate } from '@dojo/framework/core/resource';
+
+const factory = create({ icache });
+
+const animals = [
+	{ name: 'cat', label: 'cat: feline' },
+	{ name: 'dog', label: 'dog: kanine', disabled: true },
+	{ name: 'mouse', label: 'mouse: rodent' },
+	{ name: 'rat', label: 'rat: rodent' }
+];
+
+export function createMemoryTemplate<S = void>(): DataTemplate<S> {
+	return {
+		read: ({ query }, put, get) => {
```

Comment:
Is this the part you want to provide a helper for?

Reply:
yes, will need to support filtering too